x/concentrated-liquidity/swaps_test.go

```diff
@@ -560,7 +560,7 @@ var (
 			expectedTokenIn:                   sdk.NewCoin("usdc", sdk.NewInt(10000000000)),
 			expectedTokenOut:                  sdk.NewCoin("eth", sdk.NewInt(1771252)),
 			expectedFeeGrowthAccumulatorValue: sdk.MustNewDecFromStr("0.221769187794051751"),
-			expectedTick:                      sdk.NewInt(32066600),
+			expectedTick:                      sdk.NewInt(32066500),
```

Comment:
The higher the price, the greater the impact. For example, at bitcoin prices with an exponent at price one of -6 and tick spacing of 100, I believe the rounding will be in the range of [0, 0.5]. E.g. if the price is 30_000.5, it gets rounded down to 30K.

I was thinking whether this is an issue as well. The soft conclusion I have is that we, ultimately, accept this granularity as the only possible and that the users might not care about cent-level precision/rounding at the price levels of 30K. WDYT?

Reply:
To give another example,

With exponent at price one of -6 and tick spacing of 10 at the order of magnitude of 100_000, we will have a price increment of 1.

`100_000`
`100_001`

There are no prices defined between `100_000` and `100_001` because `100_000` translates to tick `45000000` and `45000010` translates to `100_001`.

If we want higher granularity, we must reduce tick spacing from 10 to 1. At that point, we have a cent-level granularity for 100K order of magnitude.

However, at 1_000_000 level exponent at price one of -6 and tick spacing of  1, we will be again in the price increment of 1.

1_000_000 is tick `54000000` and 1_000_001 is tick `54000001`. There is no 1_000_000.5 defined or possible. We cannot represent it with our tick model.

As a result, when the sqrt price goes back into the price, it might lose [0, 0.5] units of price accuracy.
For example, assume sqrt price of `1000.0002499999688` which is `math.sqrt(1_000_000.5)`.

Depending on the direction of the rounding, that will end up being converted to either tick `54000000` or tick `54000001`.
Which is either `1_000_000` or `1_000_001`.

In the worst case, the accuracy of `0.5` price units is lost (being at `1_000_000.5` and getting rounded down to `1_000_000`)

This rounding scales with the order of magnitude of the price level.

That is, 
- at 10M it is [0, 5] 
- at 100M it is [0, 50] X/Y
- at 1B it is [0, 500] X/Y

Note that I'm referring to the price and not the user position value or swap amount.

User positions are still placed on ticks, not prices and we also track sqrt price, not price. As a result, I don't see how there is any impact on user funds. This seems to be strictly about the translation between price and ticks and what prices can be used to